### PR TITLE
[Sage-496] Sage 3 - Input, Textarea, and Select

### LINKED ISSUE #496: Form fields L1 — Input and Textarea
## Description

Update component to match latest specs in Sage 3.0.

- Consider unifying as much as possible between these through mixin(s) or a common base/wrapper.
- Look ahead at the other types/variations and ensure this is open for such modifications.

**Dependencies**:
- [Border radius](https://github.com/Kajabi/sage-lib/issues/472)
- [Interactive states](https://github.com/Kajabi/sage-lib/issues/491)

**Links**

Input
- [Figma specs](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=4183%3A17166)
- [Existing component/s](https://sage-design-system.kajabi.com/pages/component/form_input)

Textarea
- [Figma specs](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=3729%3A14787)
- [Existing component/s](https://sage-design-system.kajabi.com/pages/component/form_textarea)

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description
closes #496

- [x] align form inputs and textarea

This isn't a visual difference in this work. The main difference involves updates to put common form control styles in a mixin and only added overrides where needed

## Testing in `sage-lib`

Visit pages with form elements and verify that their styles are unchanged

## Testing in `kajabi-products`

1. (LOW) No visual impact to the app

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_form_select.scss [hunk 1]
```diff
@@ -4,52 +4,46 @@
...
-  .sage-card__stack > &,
-  .sage-card__block > & {
-    margin-bottom: 0;
-  }
+  @include sage-form-field-group;
 }
 
 .sage-select__label {
-  @extend %t-sage-body;
+  @include sage-form-field-label;
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_form_select.scss:28]: 😍

File: packages/sage-assets/lib/stylesheets/components/_form_input.scss [hunk 2]
```diff
@@ -141,63 +88,18 @@ $-input-transition: border 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
...
-    color: $-input-color-error;
-  }
-}
-
-@media (prefers-reduced-motion: reduce) {
-  .sage-input__label,
-  .sage-input__field {
-    transition: none !important; /* stylelint-disable-line declaration-no-important */
-  }
+  @include sage-form-field-message;
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_form_input.scss:104]: 😍

**Review by philschanely — COMMENTED** [review]
Love where you're headed here. I think this may just need a fresh look at where the `sage-form-field` class and related modifiers are being used. Can't wait for a next round.

File: packages/sage-assets/lib/stylesheets/components/_form_input.scss [hunk 3]
```diff
@@ -141,63 +88,18 @@ $-input-transition: border 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
   }
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_form_input.scss:88]: Not a direct result of this PR, but while testing each of the updated components I noticed that hover styles are overriding the focus & active styles. See examples below where the `currentcolor` border appears above the `box-shadow`:

<details>
<summary>input</summary>
capture of input behavior
</details>

<details>
<summary>textarea</summary>
capture of textarea behavior

</details>

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM, nice refactor! Added a note about a potential follow-up item.

**Comment by monkeypox8** [comment]
One other issue that's also related to these components: #538

File: packages/sage-assets/lib/stylesheets/core/mixins/_sage.scss [hunk 4]
```diff
@@ -295,6 +295,168 @@
...
+/// Adds the basic default form label spacing, and border treatment
+///
+@mixin sage-form-field-label() {
+  @extend %t-sage-body-xsmall-semi;
+
+  position: absolute;
+  // left: $-textarea-padding;
+  // top: $-textarea-border-width;
+  // padding-left: $-textarea-label-padding;
+  // padding-right: $-textarea-label-padding;
```
Comment (FuturaExtraBold) [inline packages/sage-assets/lib/stylesheets/core/mixins/_sage.scss:426]: Is this still necessary?

**Review by FuturaExtraBold — APPROVED** [review]
Just that one nit about commented code. Really nice work here, Quinton! 👍

**Comment by FuturaExtraBold** [comment]
@QuintonJason Should we add closes #496 in the related area in the PR description?

**Pull request merged**